mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/InsertTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+package org.hibernate.ogm.datastore.mongodb.test;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+
+import org.hibernate.ogm.datastore.mongodb.utils.MongoDbServerRunner;
+import org.hibernate.ogm.utils.OgmTestCase;
+import org.hibernate.ogm.utils.TestForIssue;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author The Viet Nguyen
+ */
+@TestForIssue(jiraKey = "OGM-1222")
+@RunWith(MongoDbServerRunner.class)
+public class InsertTest extends OgmTestCase {
```

Comment:
What's the purpose of this test? We already test insertion and persisting I think

Reply:
This test was submitted by an user. He thinks there is an error if we persist then mutate the entity. I tried to reproduce, but it seems already fixed.

Reply:
Ah OK, thanks.
OK, Thanks

Reply:
Nice test. It seems the same case reported in the issue.
Could you add comment in javadoc like "Verify flush does not duplicate an entity on changing field, in the same transaction we create it". You could improve the text of the comment :P

Reply:
Maybe "Verify flush does not duplicate Entity on changing field, even if we are in the same transaction that creates the Entity itself".

Reply:
It's a minor issue... but in this way the test will be more readable from outside.